docs/src/constraints.md

```diff
@@ -235,6 +235,59 @@ The constraint can also be retrieved from its name using
 constraint_by_name
 ```
 
+## Start Values
+
+Provide a starting value (also called warmstart) for a constraint's dual using
+[`set_dual_start_value`](@ref).
+
+The start value of a constraint's dual can be queried using [`dual_start_value`](@ref).
+If no start value has been set, [`dual_start_value`](@ref) will return `nothing`.
+
+```jldoctest constraint_dual_start; setup=:(model=Model())
+julia> @variable(model, x)
+x
+
+julia> @constraint(model, con, x >= 10)
+con : x ≥ 10.0
+
+julia> dual_start_value(con)
+
+julia> set_dual_start_value(con, 2)
+
+julia> dual_start_value(con)
+2.0
+```
+
+A vector constraint will require a vector warmstart:
+
+```jldoctest constraint_dual_start_vector; setup=:(model=Model())
+julia> @variable(model, x)
+x
+
+julia> @constraint(model, con, [x, x] in SecondOrderCone())
+con : [x, x] in MathOptInterface.SecondOrderCone(2)
```

Comment:
How about `@variable(model, x[1:3])`. `@constraint(model, con, x in SecondOrderCone())`. This is a weird constraint to add.